### PR TITLE
Cs/sc 2135 hourly reports not respecting timezones

### PR DESCRIPTION
**Pull request by Charl1996**
## Product Description
No visible effects.

## Technical Summary
[Ticket](https://dimagi-dev.atlassian.net/browse/SC-2135)

When an hourly report is sent, the domain's specific timezone was not taken into consideration. This PR aims to rectify this such that the domain's specific timezone setting is taken into account when saving the ReportNotification.

[update]
Before a report is persisted in the DB, the specified day and time of the scheduled report is shifted to the server time (`soft_shift_to_server_timezone`) and before it is displayed on the UI, it is shifted to the domain's specific timezone setting (`soft_shift_to_domain_timezone`). This functionality already existed, but the only thing that is needed in addition for the `hourly` option is the `stop_hour` that also needs that conversion. 

## Feature Flag
[Add ability to send a scheduled report hourly](https://staging.commcarehq.org/hq/flags/edit/hourly-scheduled-report/)

### Safety story
Tested changes locally
Automated test for time conversion
QA testing on staging

### Automated test coverage
Test included for converting ReportNotification timezone.

### QA Plan
QA testing on staging

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by orangejenny** [comment]
> the domain's specific timezone setting is taken into account when saving the ReportNotification

I'm not close to this work, so tell me if I'm off base, but this caught my eye. It sounds like we'll now be saving the ReportNotification in terms of the project's time zone, is that true? I'd expect to save in server time, and then adjust to project time when checking if it's time to send the report - both because the project's time zone can change and because when troubleshooting cross-domain issues, it's easier if all the data is in the same time zone.

**Comment by Charl1996** [comment]
@orangejenny Oh, no, quite the opposite. 
What I meant by "the domain's specific timezone setting is taken into account when saving the ReportNotification" is that I'm simply taking the input from the user and converting that to UTC (server time) before saving (conformed the `hourly` option to what the other interval options already does, i.e. "soft_shift_to_server_timezone" before saving and "soft_shift_to_domain_timezone" before displaying). The only thing that is different with the `hourly` option is that the `stop_hour` should also be recalculated back and forth. (I'll update the description to make this more clear; looking at it now I realise this could be confusing)

**Comment by orangejenny** [comment]
@Charl1996 aha, got it - thank you for explaining

File: corehq/apps/reports/tests/test_views.py [hunk 1]
```diff
@@ -921,3 +923,55 @@ def test_non_admin_cannot_edit_other_shared_configs(self):
...
+        cls.domain_obj.default_timezone = "Pacific/Kiritimati"
+        cls.domain_obj.save()
+        super()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.domain_obj.delete()
+        super()
+
+    def convert_to_server_timezone(self):
```
Comment (sravfeyn) [inline corehq/apps/reports/tests/test_views.py:943]: If these are tests shouldn't they start with `test_`?

File: corehq/apps/reports/views.py [hunk 2]
```diff
@@ -640,6 +628,44 @@ def calculate_day(interval, day, day_change):
     return day
 
 
+def soft_shift_to_domain_timezone(instance: ReportNotification):
+    time_difference = get_timezone_difference(instance.domain)
```
Comment (sravfeyn) [inline corehq/apps/reports/views.py:632]: The only difference with `soft_shift_to_server_timezone` seems to be `recalculate_hour` and `instance.minute`. Would you consider making these DRY?

**Review by sravfeyn — APPROVED** [review]

**Pull request merged**